lib/rules/no-useless-assignment.js

```diff
@@ -309,7 +309,7 @@ module.exports = {
                 }
 
 
-                const readReferences = targetAssignment.variable.references.filter(reference => reference.isRead());
+                const readReferences = targetAssignment.variable.references.filter(reference => reference.isRead() && reference.identifier.type === "Identifier");
```

Comment:
Would this work for example from https://github.com/eslint/eslint/issues/19160#issuecomment-2499928852? I think it would be better to just skip checking anything for a variable that has at least one non-identifier reference.

Reply:
@mdjermanovic  sure 218922e